src/reformulate.jl

```diff
@@ -1,82 +1,168 @@
+################################################################################
+#                              REFORMULATE
+################################################################################
 """
-    reformulate_disjunction(m::Model, disj...; bin_var, reformulation, param)
+    reformulate_model(model::JuMP.Model, method::AbstractSolutionMethod)
 
-Reformulate disjunction on a JuMP model.
+Reformulate a `GDPModel` using the specified `method`. Prior to reformulation,
+all previous reformulation variables and constraints are deleted.
+"""
+function reformulate_model(model::JuMP.Model, method::AbstractSolutionMethod)
+    #clear all previous reformulations
+    _clear_reformulations(model)
+    #reformulate
+    _reformulate_logical_variables(model)
+    _reformulate_disjunctions(model, method)
+    _reformulate_logical_constraints(model)
+    #set solution method
+    _set_solution_method(model, method)
+    _set_ready_to_optimize(model, true)
+end
 
-    reformulate_disjunction(disj, bin_var, reformulation, param)
+function _clear_reformulations(model::JuMP.Model)
+    JuMP.delete.(model, _reformulation_constraints(model))
+    JuMP.delete.(model, _reformulation_variables(model))
+    empty!(gdp_data(model).reformulation_constraints)
+    empty!(gdp_data(model).reformulation_variables)
+end
 
-Reformulate disjunction.
-"""
-function reformulate_disjunction(m::Model, disj...; bin_var, reformulation, param)
-    #check disj
-    disj = [check_constraint!(m, constr) for constr in disj]#check_disjunction!(m, disj)
-    #get original variable refs and variable names
-    vars = setdiff(all_variables(m), m[bin_var])
-    var_names = unique(Symbol.([split("$var","[")[1] for var in vars]))
-    if !in(:gdp_variable_refs, keys(object_dictionary(m)))
-        @expression(m, gdp_variable_refs, vars)
-    end
-    if !in(:gdp_variable_names, keys(object_dictionary(m)))
-        @expression(m, gdp_variable_names, var_names)
-    end
-    #run reformulation
-    if reformulation == :hull
-        disaggregate_variables(m, disj, bin_var)
-        sum_disaggregated_variables(m, disj, bin_var)
+################################################################################
+#                              LOGICAL VARIABLES
+################################################################################
+# create binary (indicator) variables for logic variables.
+function _reformulate_logical_variables(model::JuMP.Model)
+    for (lv_idx, lv_data) in _logical_variables(model)
+        lv = lv_data.variable
+        lvref = LogicalVariableRef(model, lv_idx)
+        bvref = JuMP.@variable(model, base_name = lv_data.name, binary=true, start = lv.start_value)
```

Comment:
Does this work if `lv.start_value = nothing`?

Reply:
Yes, `nothing` is the default start value on a logical variable